sdk/nodejs/runtime/settings.ts

```diff
@@ -539,15 +539,25 @@ function options(): Options {
  * Permanently disconnects from the server, closing the connections. It waits
  * for the existing RPC queue to drain.  If any RPCs come in afterwards,
  * however, they will crash the process.
- */
-export function disconnect(): Promise<void> {
-    return waitForRPCs(/*disconnectFromServers*/ true);
+ *
+ * If `signalShutdown` is true, signal to the monitor that we're ready to
+ * shutdown. This will wait until the monitor has completed all steps,
+ * including deletion steps.
+ */
+export async function disconnect(signalShutdown = false): Promise<void> {
+    await waitForRPCs();
+    if (signalShutdown) {
+        await signalAndWaitForShutdown();
+    }
+    disconnectSync();
 }
 
 /**
+ * Waits for the existing RPC queue to drain.
+ *
  * @internal
  */
-export function waitForRPCs(disconnectFromServers = false): Promise<void> {
+export function waitForRPCs(): Promise<void> {
```

Comment:
We only ever pass `disconnectFromServers=true` in `disconnect` above, so I moved `disconnectSync` call up into that function.